students/mark/Session01/break_me.py

```diff
@@ -0,0 +1,163 @@
+#!/usr/local/bin/python3
+
+import sys
+
+
+"""Explore Errors
+
+    https://uwpce-pythoncert.github.io/PythonCertDevel/exercises/python_pushups.html#python-pushups
+
+    Add a new file to it called break_me.py
+
+    In the break_me.py file write four simple Python functions:
+
+    Each function, when called, should cause an exception to happen
+    Each function should result in one of the four most common exceptions you’ll find.
+    for review: NameError, TypeError, SyntaxError, AttributeError
+
+    NameError, TypeError, SyntaxError, AttributeError
+"""
+
+
+def simpleNameError():
+    """Example to raise a type NameError"""
+    """Call an function with a value not defined"""
+
+    raise NameError('Raising a NameError')
+
+
+def altNameError():
+    print(yourstuffisbroke)
+
+
+def simpleTypeError():
+    """Example to raise a type TypeError"""
+
+    raise TypeError('Raising a TypeError')
+
+
+def altTypeError():
+    """Example to raise a type TypeError"""
+
+    notANumber = "1"
+    total = notANumber + 3 
+
+    return total
+
+
+def simpleSyntaxError():
+    """Example to raise a type SyntaxError"""
+
+    raise SyntaxError('Raising a SyntaxError')
+
+
+def altSyntaxError():
+    """Alt Example to raise a type SyntaxError"""
+
+    ### Note: uncomment for syntax error this does not even "compile"
+
+
+    ## altSyntaxError_ThisIsASyntaxError():
+
+    """
+     Thoughts and options:
+     Maybe wrap in try except block and toss a continue later????
+     Nope, that fails on "compile" too.
+     There has got to be an easier way: https://docs.python.org/3/library/warnings.html
+     The python3 interpreter is really good at letting you know there is a sytnax error.
+    """
```

Comment:
Isn't it amazing how much harder it is to throw an error when you're actually trying to? :)